dask_ml/metrics/regression.py

```diff
@@ -88,7 +108,20 @@ def r2_score(y_true, y_pred, sample_weight=None,
     output_scores[valid_score] = 1 - (numerator[valid_score] /
                                       denominator[valid_score])
     output_scores[nonzero_numerator & ~nonzero_denominator] = 0.
-    result = output_scores.mean(axis=0)
+
+    if isinstance(multioutput, six.string_types):
+        if multioutput == 'raw_values':
+            # return scores individually
+            return output_scores.compute() if compute else output_scores
```

Comment:
Why the compute call here and not elsewhere?  

Please forgive my ignorance, but what are our current policies around automatically computing scores vs letting the user call compute themselves?

Reply:
I'll need to verify, I believe the default `compute=True` came out of trying to get sklearn.model_selection.GridSearchCV to work well with Incremental.

Initially, I had it lazy, but something failed. Apologies for not having logged the output of what happens when scikit-learn receives a scalar dask array as a score, and for not digging further on it.

Reply:
@TomAugspurger you're right about `sklearn.model_selection.GridSearchCV` not working well with scalar dask arrays as a score. The example snippet below is fine when `compute=True`, but fails when `compute=False`

```python
from functools import partial

from dask_ml.datasets import make_classification
from dask_ml.wrappers import Incremental
from dask_ml.metrics import accuracy_score

from sklearn.metrics import make_scorer
from sklearn.linear_model import SGDClassifier
from sklearn.model_selection import GridSearchCV

# Get dataset
X, y = make_classification(chunks=25)

estimator = SGDClassifier(random_state=10)
acc = partial(accuracy_score, compute=False)
acc.__name__ = 'accuracy_score' # make_scorer looks for __name__
clf = Incremental(estimator, scoring=make_scorer(acc))

gs = GridSearchCV(clf, param_grid={'estimator__loss': ['hinge', 'log']})
gs.fit(X, y, classes=[0, 1])
```

When `compute=False`, calling `GridSearchCV.fit` raises 

```
ValueError: scoring must return a number, got dask.array<mul, shape=(), dtype=float64, chunksize=()> (<class 'dask.array.core.Array'>) instead. (scorer=score)
```

which comes from https://github.com/scikit-learn/scikit-learn/blob/30ebb98dff5a7773ac66d068c8103cd47c0427ec/sklearn/model_selection/_validation.py#L606-L609